uPortal-events/src/main/java/org/apereo/portal/events/PortalEventFactoryImpl.java

```diff
@@ -561,6 +568,24 @@ public void publishTenantRemovedTenantEvent(
         this.applicationEventPublisher.publishEvent(event);
     }
 
+    /*
+     * Analytics Events
+     */
+
+	@Override
+	public void publishAnalyticsPortalEvents(HttpServletRequest request, Object source, Map<String, Object> analyticsData, IPerson person) {
+        final PortalEvent.PortalEventBuilder eventBuilder =
+                this.createPortalEventBuilder(source, request);
+        Date eventDate = Calendar.getInstance(TimeZone.getTimeZone("UTC")).getTime();
+        String rawEventType = (String) analyticsData.computeIfAbsent("type", key -> "UNKNOWN");
+        // throws IllegalArgumentException
+        String eventType = InputValidator.validateAsWordCharacters(rawEventType, "String");
```

Comment:
I would change the second parameter to 'event type' to help end users locate the issue with the failed request.